src/qat/purr/backends/utilities.py

```diff
@@ -152,34 +156,33 @@ def __init__(self, std_dev, width, zero_at_edges):
     @validate_input_array
     def eval(self, x: np.ndarray) -> np.ndarray:
         zae_chunk = self.zero_at_edges * (
-            np.exp(-0.5 * (self.width / 2 * self.std_dev) ** 2)
+            np.exp(-0.5 * ((self.width / 2) / self.std_dev) ** 2)
         )
         coef = 1 / (1 - zae_chunk)
         gauss = np.exp(-0.5 * (x / self.std_dev) ** 2)
         return coef * (gauss - zae_chunk)
 
 
-class GaussianSquare(ComplexFunction):
-    def __init__(self, width, square_width):
-        self.width = width
+class GaussianSquareFunction(NumericFunction):
+    """
+    A square pulse with a Gaussian rise and fall at the edges.
+    """
+
+    def __init__(self, square_width: float, std_dev: float, zero_at_edges: int):
         self.square_width = square_width
+        self.std_dev = std_dev
+        self.zero_at_edges = zero_at_edges
 
     @validate_input_array
     def eval(self, x: np.ndarray) -> np.ndarray:
-        gauss = np.exp(-0.5 * (x / self.width) ** 2)
-        square = np.ones(shape=x.shape, dtype=self._dtype)
-        square_slots = int(self.square_width * self.dt)
-        if square_slots > len(x):
-            raise ValueError(
-                f"The length of the square portion [[{self.square_width}]] cannot "
-                f"exceed the total pulse length [[{len(x)*self.dt}]] in a square "
-                "guassian."
-            )
-        gaus_len = int((len(x) - square_slots) / 2)
-        first_chunk = gauss[0:gaus_len]
-        second_chunk = square[0:square_slots]
-        final_chunk = gauss[gaus_len + square_slots :]
-        return append(first_chunk, second_chunk, final_chunk)
+        y = np.ones(shape=x.shape, dtype=self._dtype)
+        x_rise = x[x < -self.square_width / 2] + (self.square_width / 2)
+        x_fall = x[x > self.square_width / 2] - (self.square_width / 2)
+        y[x < -self.square_width / 2] = np.exp(-0.5 * (x_rise / self.std_dev) ** 2)
+        y[x > self.square_width / 2] = np.exp(-0.5 * (x_fall / self.std_dev) ** 2)
+        if self.zero_at_edges != 0:
```

Comment:
using `zero_at_edges` as boolean here, can't you just specify it as a boolean argument ?